code/modules/library/library_catalog.dm

```diff
@@ -0,0 +1,616 @@
+	)
+
+	//building a list of all categories, used for searching, Cat ID needs to be unique to category
+	categories = list(
+		DEFINE_CATEGORY(LIB_CATEGORY_FICTION, "Fiction"),
+		DEFINE_CATEGORY(LIB_CATEGORY_NONFICTION, "Non-Fiction"),
+		DEFINE_CATEGORY(LIB_CATEGORY_RELIGION, "Religious"),
+		DEFINE_CATEGORY(LIB_CATEGORY_FANTASY, "Fantasy"),
+		DEFINE_CATEGORY(LIB_CATEGORY_HORROR, "Horror"),
+		DEFINE_CATEGORY(LIB_CATEGORY_ROMANCE, "Romance"),
+		DEFINE_CATEGORY(LIB_CATEGORY_MYSTERY, "Mystery"),
+		DEFINE_CATEGORY(LIB_CATEGORY_ADVENTURE, "Adventure"),
+		DEFINE_CATEGORY(LIB_CATEGORY_HISTORY, "History"),
+		DEFINE_CATEGORY(LIB_CATEGORY_PHILOSOPHY, "Philosophy"),
+		DEFINE_CATEGORY(LIB_CATEGORY_DRAMA, "Drama and Thriller"),
+    	DEFINE_CATEGORY(LIB_CATEGORY_EXPERIMENT, "Experiment Notes"),
+		DEFINE_CATEGORY(LIB_CATEGORY_LEGAL, "Legal Document"),
+		DEFINE_CATEGORY(LIB_CATEGORY_BIOGRAPHY, "Biography"),
+		DEFINE_CATEGORY(LIB_CATEGORY_GUIDE, "Guides and References"),
+		DEFINE_CATEGORY(LIB_CATEGORY_PAPERWORK, "Paperwork"),
+		DEFINE_CATEGORY(LIB_CATEGORY_COOKING, "Culinary Arts"),
+		DEFINE_CATEGORY(LIB_CATEGORY_DESIGN, "Decor and Design"),
+		DEFINE_CATEGORY(LIB_CATEGORY_COMBAT, "Martial Arts and Combat"),
+		DEFINE_CATEGORY(LIB_CATEGORY_EXPLORATION, "Exploration"),
+		DEFINE_CATEGORY(LIB_CATEGORY_THEATRE, "Theatre"),
+		DEFINE_CATEGORY(LIB_CATEGORY_POETRY, "Poetry"),
+    	DEFINE_CATEGORY(LIB_CATEGORY_LAW, "Law"),
+		DEFINE_CATEGORY(LIB_CATEGORY_SECURITY, "Security"),
+		DEFINE_CATEGORY(LIB_CATEGORY_SUPPLY, "Supply"),
+		DEFINE_CATEGORY(LIB_CATEGORY_ENGINEERING, "Engineering"),
+		DEFINE_CATEGORY(LIB_CATEGORY_SERVICE , "Service"),
+		DEFINE_CATEGORY(LIB_CATEGORY_MEDICAL, "Medical"),
+		DEFINE_CATEGORY(LIB_CATEGORY_RESEARCH, "Science"),
+		DEFINE_CATEGORY(LIB_CATEGORY_COMMAND , "Command"),
+	)
+
+	//Books that we don't want showing up in the programmatic book list
+	//Books should go here if they're non-functional, spawners, or are designed for off-station roles to consume
+	var/list/forbidden_books = list(
+		/obj/item/book/manual/random,
+		/obj/item/book/manual/nuclear,
+		/obj/item/book/manual/wiki,
+		/obj/item/book/manual/hydroponics_pod_people,
+	)
+
+	var/newid = 1
+	//building a list of all programmatic books
+	for(var/typepath in (subtypesof(/obj/item/book/manual) - forbidden_books))
+		var/obj/item/book/B = new typepath(null)
+		var/datum/programmaticbook/PB = new()
+		PB.title = B.name
+		PB.author = B.author
+		PB.id = "M[newid]"
+		PB.path = typepath
+		newid++
+		books += PB
+
+/*
+ * can_vv_delete override
+ * Admins should not be deleting this willy nilly, if they think it is neccesary,
+ * they can go through the effort of advanced proccall
+ */
+/datum/library_catalog/can_vv_delete()
+	message_admins("An admin attempted to VV delete the global library catalog, this will break the library system for the round, if you know what you are doing please use advanced proccal")
+	return FALSE
```

Comment:
[Out of PR scope].

I want to make a `SHOULD_NOT_DEL` datum flag at some point which handles wrapping vvdel and /Destroy() stuff.

Reply:
Yeah, I wasn't sure 100% how I wanted to implement something like this considering this is only 1 of a handful of implementations where can_vv_delete() is used. I just wanted to make sure 100% that this never gets deleted accidently.